app.py

```diff
@@ -1,14 +1,57 @@
 #!/usr/bin/env python3
+"""
+This stack generates the necessary AWS services for the data management 
+portions of a Science Data System.  
+
+Options can be included by adding "--context param=value" to a cdk s
+ynth/deploy command, for example:
+
+cdk deploy --context SDSID=harter-testing \
+           --context userpool_name=sds-userpool-dev \
+           --context app_client_name=sdscommandline-dev \
+           --all
+
+:context SDSID: Required. This is a unique identifier for your stack.  
+:context initial_user: Optional. This is the email address of the initial
+                       cognito user.  If not provided, the first user will 
+                       need to be set up in the AWS console.   
+:context userpool_name: Optional (Required if app_client_name given). 
+                        This is the name of the userpool that contains 
+                        the specified app_client_name.
+:context app_client_name: Optional (Required if userpool_name given). 
+                          This is the name of the app client on the AWS 
+                          account that will be used to verify received 
+                          tokens in the API.  
+:context cognito_only: Optional.  If true, then this stack will only 
+                       deploy cognito services.  
+
+To summarize, there are 3 "modes" for this application:
+
+1) Cognito is spun up at the same time as the rest of the stack.  
+   Context needed: None  
+2) No Cognito services are created and a pre-existing cognito userpool is used.  
+   Context needed: "userpool_name" and "app_client_name".  
+3) ONLY Cognito services are created.  
+   Context needed: "cognito_only".   
+
+"""
 import random
 import string
 
 import aws_cdk as cdk
+import boto3
 
+from sds_data_manager.sds_cognito_stack import SdsCognitoStack
 from sds_data_manager.sds_data_manager_stack import SdsDataManagerStack
 
 app = cdk.App()
 
+# Grab context from cdk synth and cdk deploy commands
 SDS_ID = app.node.try_get_context("SDSID")
+initial_user = app.node.try_get_context("initial_user")
+userpool_name = app.node.try_get_context("userpool_name")
+app_client_name = app.node.try_get_context("app_client_name")
+cognito_only = app.node.try_get_context("cognito_only")
```

Comment:
This seems like it is adding a lot of complication and confusion for deployment. We'll now have to remember all of these options if we want to deploy something.

Reply:
If we do put the cognito in a separate stack, how would we tell the data-management stack what the userpool/app_clients are? Would they be hard-coded?